src/eth/miner/miner.rs

```diff
@@ -472,11 +473,24 @@ mod interval_miner {
     pub fn run(miner: Arc<Miner>, ticks_rx: mpsc::Receiver<Instant>) {
         const TASK_NAME: &str = "interval-miner-ticker";
 
-        while let Ok(tick) = channel_read_sync!(ticks_rx) {
+        loop {
             if GlobalState::is_shutdown_warn(TASK_NAME) {
-                return;
+                break;
             }
 
+            let tick = match ticks_rx.recv_timeout(Duration::from_secs(2)) {
+                Ok(tick) => tick,
+                Err(RecvTimeoutError::Disconnected) => break,
+                Err(RecvTimeoutError::Timeout) => {
+                    tracing::warn!(
+                        channel = stringify!(ticks_rx),
```

Comment:
Mmmm this log might not be necessary anymore, since we now have a custom log message (2 lines below).